### PR TITLE
test case for #11367

### PR DESCRIPTION
**Pull request by rrd108**
Refs https://github.com/cakephp/cakephp/issues/11367

### CODE REVIEW

File: tests/TestCase/Database/QueryTest.php [hunk 1]
```diff
@@ -403,6 +405,37 @@ public function testSelectRightJoin()
...
+            'connection' => $this->connection
+        ]);
+
+        $latest = $query->select([
+                'u_id' => 'Comments.user_id',
+                'latest' => $query->func()->max('Comments.created')
+            ])->from(['Comments' => 'comments'])
+            ->group('Comments.user_id');
+
+        $actual = $comments->find()->rightJoin(
```
Comment (markstory) [inline tests/TestCase/Database/QueryTest.php]: Why is the right join relevant?

Comment (rrd108) [reply tests/TestCase/Database/QueryTest.php]: In my original code I had a right join. I just kept it in the test.

Comment (markstory) [reply tests/TestCase/Database/QueryTest.php]: Ok. I've posted a comment on #11367 as I think you use case is not generating the correct SQL for the join conditions as right side values are treated as bound parameters.

**Comment by ADmad** [comment]
Closing as the provided test case isn't support to work.

**Pull request closed without merging**